### DISCUSSION TITLE
Some ideas for moving forward

### DISCUSSION THREAD

**Discussion by navidcy** [jbisits/QGTracerAdvection#15] [General]
@jbisits, since the scripts now "do the job" for you, perhaps you can go ahead move forward and do some science/physics explorations with them.

I'll try to make the code work even more smoothly with incorporating `GeophysicalFlows.jl` problem into `PassiveTracerFlows.TracerAdvectionDiffusion` module. I suggest we leave a copy of the module you wrote + `test_TrAdvDiff_QG.jl` intact (perhaps I'll just create a new branch and don't modify it) so that, when I go ahead and make changes I can benchmark it to the code there exist now to make sure I don't break anything.

(FYI: I won't find much time to do so before Fri/Sat...)

**Comment by navidcy** [comment]
Here's the branch: https://github.com/jbisits/QG_tracer_advection/tree/ncc/archive-TracerAdvDiff_QG-module

**Comment by jbisits** [comment]
@navidcy, yep the plan is to do exactly that! Another thing I will add to the module (hopefully sooner rather than later) is a function that can "drop" the tracer in to the flow at a time prescribed by the user. To do this I will most likely create another input (time to drop in tracer), run the `MultiLayerQG.Problem` till it reaches the desired time, initialise the velocity field in the `TracerAdvDiff_QG` then runs the main loop as usual from there. I will do this in a separate pull request which will leave the current module intact. 

What you have done and helped me with is GREAT! Thank you very much!!!!!

**Comment by navidcy** [comment]
@jbisits, this is not really an issue. I suggest enabling the Discussions from the repo settings (I can't do that) and then convert this issue to a Discussion.

<img width="800" alt="Screen Shot 2021-05-13 at 4 11 42 pm" src="https://user-images.githubusercontent.com/7112768/118085618-0ca6e600-b406-11eb-8d06-c8c3f266c1f2.png">

<img width="800" alt="Screen Shot 2021-05-13 at 4 12 13 pm" src="https://user-images.githubusercontent.com/7112768/118085651-16304e00-b406-11eb-8ab7-e3f20366427e.png">

**Reply by jbisits** [reply]
Done!

**Reply by jbisits** [reply]
I just put it in the general folder but maybe ideas is better
